src/main/java/com/puppetlabs/jenkins/plugins/puppetgatling/SimulationMetrics.java

```diff
@@ -0,0 +1,82 @@
+import hudson.FilePath;
+import hudson.model.Run;
+import io.gatling.jenkins.chart.Point;
+import io.gatling.jenkins.chart.Serie;
+import io.gatling.jenkins.chart.SerieName;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class SimulationMetrics {
+
+    private static final int MAX_MEMORY_DATA_POINTS_TO_DISPLAY = 40;
+    private static final int BYTES_PER_MEGABYTE = (1024 * 1024);
+
+    private final Graph<Long> memoryUsage;
+
+    public SimulationMetrics(Run<?, ?> run, PrintStream logger, FilePath workspace,
+                             String simulationId) throws IOException, InterruptedException {
+
+        logger.println("Checking for existence of metrics.json.");
+        FilePath metricsFilePath = workspace.child("puppet-gatling").
+                child(simulationId).child("sut_archive_files").
+                child("metrics.json");
+        if (metricsFilePath.exists()) {
+            logger.println("Found metrics.json, parsing.");
+            List<Point<Integer, Long>> memoryData = new ArrayList<>();
+
+            InputStream metricsInput = metricsFilePath.read();
+            JsonFactory jsonFactory = new JsonFactory();
+            JsonParser jsonParser = jsonFactory.createParser(metricsInput);
+            jsonParser.setCodec(new ObjectMapper());
+            jsonParser.nextToken();
+
+            int i = 0;
+            while (jsonParser.hasCurrentToken()) {
+                Map metricsEntry = jsonParser.readValueAs(Map.class);
+                Long usedHeapValue = getValueFromNestedMap(metricsEntry,
+                        new String[] {"status-service", "status", "experimental", "jvm-metrics", "heap-memory", "used"}
+                );
+                i++;
+                memoryData.add(new Point<Integer, Long>(i, usedHeapValue / BYTES_PER_MEGABYTE));
+                jsonParser.nextToken();
+            }
+            metricsInput.close();
+
+            SerieName memSeriesName = new SerieName("memory");
+            Map<SerieName, Serie<Integer, Long>> fakeData = new TreeMap<>();
+            logger.println("Found " + memoryData.size() + " memory data points.");
+            fakeData.put(memSeriesName, RawDataGraph.filterDataToSeries(memoryData, MAX_MEMORY_DATA_POINTS_TO_DISPLAY));
+
+            memoryUsage = new RawDataGraph<Long>(fakeData);
+        } else {
+            logger.println("No metrics.json found; memory data will not be visible.");
+            memoryUsage = null;
+        }
+    }
+
+    private static Long getValueFromNestedMap(Map firstEntry, String[] strings) {
+        Map next = firstEntry;
+        for (int i = 0; i < strings.length - 1; i++) {
+            String key = strings[i];
+            next = (Map) next.get(key);
+        }
+        return (Long) next.get(strings[strings.length - 1]);
```

Comment:
When I ran the updated plugin on a test Jenkins node, I got the following error:

````
Checking for existence of metrics.json.
Found metrics.json, parsing.
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.Long
	at com.puppetlabs.jenkins.plugins.puppetgatling.SimulationMetrics.getValueFromNestedMap(SimulationMetrics.java:75)
	at com.puppetlabs.jenkins.plugins.puppetgatling.SimulationMetrics.<init>(SimulationMetrics.java:49)
	at com.puppetlabs.jenkins.plugins.puppetgatling.PuppetGatlingPublisher.getBuildAction(PuppetGatlingPublisher.java:156)
	at com.puppetlabs.jenkins.plugins.puppetgatling.PuppetGatlingPublisher.perform(PuppetGatlingPublisher.java:106)
	at com.puppetlabs.jenkins.plugins.puppetgatling.steps.PuppetGatlingArchiverStepExecution.run(PuppetGatlingArchiverStepExecution.java:30)
	at com.puppetlabs.jenkins.plugins.puppetgatling.steps.PuppetGatlingArchiverStepExecution.run(PuppetGatlingArchiverStepExecution.java:11)
	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1$1.call(AbstractSynchronousNonBlockingStepExecution.java:49)
	at hudson.security.ACL.impersonate(ACL.java:213)
	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1.run(AbstractSynchronousNonBlockingStepExecution.java:47)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Finished: FAILURE
````

Maybe the metric could be an Integer or Long depending upon how large the value was in the JSON file.  I'm experimenting with the following code which tries to handle coercing to a Long differently if the original value is an Integer although there's probably a better way to resolve this?

```java
Object value = next.get(strings[strings.length - 1]);
return value.getClass() == Integer.class ? ((Integer) value).longValue() : (Long) value;
```